### PR TITLE
Add storybook to Svelte TS project

### PR DESCRIPTION
**Pull request by gcsecsey**
# Pull Request Template

## Description

This PR adds Storybook with Typescript support to the `agnostic-svelte-ts` project:
- Stories can now use the [Svelte native format](https://storybook.js.org/blog/storybook-for-svelte/#svelte-native-format
)
- The `copystyles.cjs` file and the npm script invoking it is copied over to the project, but currently throws an error because of missing components
- Prettier config is updated to use the SvelteKit defaults as well as settings from `agnostc-svelte`
- The Alert component is copied with some type errors fixed and the new story format

Related to #127

## Testing steps

```
cd agnostic-svelte-ts
npm i && npm run storybook
```

Storybook should build all stories with TS sources and common CSS successfully:
image

## Checklist:

Please delete options that are not relevant.

- [X] Is this a new feature
- [X] Have you submitted new feature or bugfix issue? Unless it's a small bugix, we generally prefer to track PRs against an issue which starts a dialogue to give us context.

_We realize above checklist is pretty intimidating, reach out with an at mention on your issue if you're interested in contributing but need some clarifications!_

### CODE REVIEW

File: agnostic-svelte-ts/.prettierrc.cjs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+	useTabs: true,
+	singleQuote: true,
+	trailingComma: 'none',
```
Comment (roblevintennis) [inline agnostic-svelte-ts/.prettierrc.cjs:4]: Non-blocking: I actually prefer trailing commas but not a blocker haha

File: agnostic-svelte-ts/.storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+			date: /Date$/
+		}
+	}
+};
+
+export const decorators = [() => FontDecorator];
+
+require('!style-loader!css-loader!../src/lib/css/common.resets.min.css');
+require('!style-loader!css-loader!../src/lib/css/common.properties.min.css');
+require('!style-loader!css-loader!../src/lib/css/common.utilities.min.css');
```
Comment (roblevintennis) [inline agnostic-svelte-ts/.storybook/preview.js:22]: Are these actually working properly in SB? I've historically found SB to be a real PITA for these globals and tracking latest. This is awesome if it's working ❤️

File: agnostic-svelte-ts/.storybook/yourTheme.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { create } from '@storybook/theming/create';
+
+export default create({
+	base: 'light',
+
+	// blue primary
```
Comment (roblevintennis) [inline agnostic-svelte-ts/.storybook/yourTheme.js:6]: nice touch!

File: agnostic-svelte-ts/src/lib/components/Alert/Alert.stories.svelte [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	import { Meta, Template, Story } from '@storybook/addon-svelte-csf';
+</script>
+
+<Meta title="AG—Svelte (Beta)/Alerts" component="{Alert}" />
+
+<Template let:args>
+	<Alerts {...args} />
+</Template>;
+
+<Story name="Deafult" />;
```
Comment (roblevintennis) [inline agnostic-svelte-ts/src/lib/components/Alert/Alert.stories.svelte:13]: Default

File: agnostic-svelte-ts/src/lib/components/Alert/Alert.svelte [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+	@media (prefers-reduced-motion), (update: slow) {
+		.slide-up-fade-in,
+		.fade-in,
+		.slide-up {
+			transition-duration: 0.001ms !important;
+		}
+	}
+</style>
+
+<script lang="ts">
```
Comment (roblevintennis) [inline agnostic-svelte-ts/src/lib/components/Alert/Alert.svelte:241]: Woohoo 🙌

File: agnostic-svelte-ts/src/lib/index.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import Alert from '$lib/components/Alert/Alert.svelte';
```
Comment (roblevintennis) [inline agnostic-svelte-ts/src/lib/index.js:1]: Will this `index.js` file become `index.ts` at some point?

**Comment by roblevintennis** [comment]
Notes

- Was able to comment out everything but the Alert from `agnostic-svelte-ts/copystyles.cjs` and successfully run `npm run syncStyles` and it copies over said styles into `<style>...</style>`
- Ran `npm i` and then `npm run storybook`:
image

I think this all checks out. Thanks @gcsecsey -- I'm going to merge it. Only feedback are nitpicks which you can elect to incorporate into future work if you agree ;) 🙏 ❤️ 🚀

File: agnostic-svelte-ts/src/lib/components/Alert/Alert.stories.svelte (hunk 4 above)
Comment (roblevintennis) [reply agnostic-svelte-ts/src/lib/components/Alert/Alert.stories.svelte:13]: Don't bother I've already patched this ;)

File: agnostic-svelte-ts/.prettierrc.cjs (hunk 1 above)
Comment (gcsecsey) [reply agnostic-svelte-ts/.prettierrc.cjs:4]: Yeah I also couldn't really decide here, I personally prefer trailing commas too. 😄 On one hand the SvelteKit installer generated the 'none' rule, on the other I looked at the other projects, and all of them used trailing commas, so I went with the SvelteKit default. I think it's a good question in general whether the subprojects should have the exact same formatting ruleset or not?

File: agnostic-svelte-ts/src/lib/index.js (hunk 6 above)
Comment (gcsecsey) [reply agnostic-svelte-ts/src/lib/index.js:1]: Oops forgot to update the extension when adding it, sorry! As it only has imports and exports, I think we can safely update this any time. 🙂

File: agnostic-svelte-ts/.storybook/yourTheme.js (hunk 3 above)
Comment (gcsecsey) [reply agnostic-svelte-ts/.storybook/yourTheme.js:6]: Thanks but I just brought over the config from the svelte project 😄

File: agnostic-svelte-ts/.storybook/preview.js (hunk 2 above)
Comment (gcsecsey) [reply agnostic-svelte-ts/.storybook/preview.js:22]: Yes, they do! 🎉 I've tested adding a utility class (eg. `.h1` or `.m2`) to the Alert component and it seems that SB correctly bundles these styles! I've also saw that `app.css` was imported here in the svelte project, even though it didn't contain any additional rules. I've tested that that's still working in the TS project as well.

File: agnostic-svelte-ts/.prettierrc.cjs (hunk 1 above)
Comment (roblevintennis) [reply agnostic-svelte-ts/.prettierrc.cjs:4]: Good point. I think they may have different idiomatic conventions

**Pull request merged**